contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -5864,3 +5870,44 @@ is_bbf_db_ddladmin_operation(Oid namespaceId)
 
 	return false;
 }
+
+/*
+ * remove_db_name_in_schema - remove the db name and underscore at the beginning
+ * 	of the given string. It is used to unmap schema name in error messages.
+ *
+ * 	@param schema_name - char *
+ * 	@return - unmapped schema name char *
+ */
+static const char *
+remove_db_name_in_schema(const char *object_name)
+{
+	const char	*cur_db_name;
+	char		**splited_object_name;
+	char		*schema_name = NULL;
+	char		*mutable_name;
+    size_t 		db_name_len;
+    size_t 		prefix_len;
+	size_t 		schema_name_len;
+
+	mutable_name = strdup(object_name);
+	splited_object_name = split_object_name(mutable_name);
+	for (int i = 0; i < 4; i++)
```

Comment:
is it guaranteed to have 4 element in that list? can you check the length of that list first?

Reply:
Yes, the function returns 4 element list

Reply:
can you double check whether the elements in `splited_object_name` need to be freed